src/probnum/filtsmooth/gaussfiltsmooth/kalman.py

```diff
@@ -283,3 +342,42 @@ def smooth_step(self, unsmoothed_rv, smoothed_rv, start, stop, **kwargs):
             + smoothing_gain @ (smoothed_rv.cov - predicted_rv.cov) @ smoothing_gain.T
         )
         return Normal(new_mean, new_cov)
+
+    def _smooth_step_with_preconditioning(
+        self, unsmoothed_rv, smoothed_rv, start, stop, intermediate_step=None
+    ):
+        """
+        A single smoother step.
+
+        Consists of predicting from the filtering distribution at time t
+        to time t+1 and then updating based on the discrepancy to the
+        smoothing solution at time t+1.
+
+        Parameters
+        ----------
+        unsmoothed_rv : RandomVariable
+            Filtering distribution at time t.
+        smoothed_rv : RandomVariable
+            Prediction at time t+1 of the filtering distribution at time t.
+        start : float
+            Time-point of the to-be-smoothed RV.
+        stop : float
+            Time-point of the already-smoothed RV.
+        """
+        # It is not clear to me how to best test this, except running IBM smoothing for high-ish order. (N)
```

Comment:
This. 

It is tested, because IBM-ode smoothing is tested, though I am not sure whether this is all we want.

Reply:
What about something like comparing `_smooth_step_with_preconditioning` to `_smooth_step`, first for a case in which both should output the same, and then for a case in which the latter fails but the first succeeds?

Though I'm not sure if that actually adds something compared to running a high-order low-tolerance IBM..

Reply:
It certainly does, though it might be extremely granular? I dont know, maybe we open an issue that this is not quite tested in an isolated way? Dont have to though

Reply:
Your call on what to do here, I don't have a really good overview of the current test coverage :/